packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -616,8 +616,42 @@ const PostsPage = ({fullPost, postPreload, eagerPostComments, refetch, classes}:
 
   const marketInfo = getMarketInfo(post)
 
+  // check for deep equality between terms and eagerPostComments.terms
+  const useEagerResults = eagerPostComments && isEqual(commentTerms, eagerPostComments?.terms);
+
+  const lazyResults = useMulti({
+    terms: {...commentTerms, postId: post._id},
+    skip: useEagerResults,
+    ...postsCommentsThreadMultiOptions,
+  });
+
+  const { loading, results, loadMore, loadingMore, totalCount } = useEagerResults ? eagerPostComments.queryResponse : lazyResults;
+
+  const commentCount = results?.length ?? 0;
+  const commentTree = unflattenComments(results ?? []);
+  const answersTree = unflattenComments(answersAndReplies ?? []);
+
+  const hashCommentId = location.hash.length >= 1 ? location.hash.slice(1) : null;
+  // If the comment reference in the hash doesn't appear in the page, try and load it separately as a permalinked comment
+  const showHashCommentFallback = useMemo(() => (
+    hashCommentId && !loading && results && !results.map(({ _id }) => _id).includes(hashCommentId)
+  ), [hashCommentId, loading, results]);
```

Comment:
I noticed that linking to [this comment](http://localhost:3000/posts/xafpj3on76uRDoBja/the-ftx-future-fund-team-has-resigned-1#RCJxuayttms8qahjo) didn't work. I think maybe it's because the comment is loaded, but it's collapsed or not rendered or something?

If you go to [this link on prod](https://forum.effectivealtruism.org/posts/xafpj3on76uRDoBja/the-ftx-future-fund-team-has-resigned-1?commentId=RCJxuayttms8qahjo), then the comment does load at the top.

Reply:
Good catch, it was because the comment starts collapsed. I have made it uncollapse in a `useEffect` if any of the children match the comment hash in the url